src/globals.ts

```diff
@@ -83,22 +87,27 @@ export class SpansContainer {
     incrementTotalSpansCounter: boolean = true,
     numOfDroppedSpans: number = 1
   ): void {
-    this.droppedSpansReasons[reason] =
-      (this.droppedSpansReasons[reason.valueOf()] || 0) + numOfDroppedSpans;
+    // Check if the reason exists in the droppedSpansReasons object
+    // If it doesn't exist, initialize
+    if (!this.droppedSpansReasons[reason.valueOf()]) {
+      this.droppedSpansReasons[reason.valueOf()] = { drops: 0 };
+    }
+    this.droppedSpansReasons[reason.valueOf()].drops += numOfDroppedSpans;
```

Comment:
i think it's weird to use "valueOf()". why not just use the enum as the key of the map?
but anyway, not a big deal

Reply:
Your right, fixed it